File: Lista1/classes/Point2D/Point2D.cpp
#include "Point2D.h"

// Default constructor for point 2D
Point2D::Point2D() {}

// Create point 2D
Point2D::Point2D(int posX, int posY) {
    x = posX;
    y = posY;
}

// Return position X of point 2D
int Point2D::getX() {
    return x;
}

// Return position Y of point 2D
int Point2D::getY() {
    return y;
}

// Print point 2D information
void Point2D::printPoint() {
    std::cout << "Point 2D" << std::endl;
    std::cout << "X = " << getX() << std::endl;
    std::cout << "Y = " << getY() << std::endl;
}

File: Lista1Java/src/Point2d.java
/**
 * Class to creating 2D points
 */
public class Point2d {
    /** Position X */
    protected int x;

    /** Position Y */
    protected int y;

    public Point2d() { }

    public Point2d(int posX, int posY) {
        this.x = posX;
        this.y = posY;
    }

    /**
     * Return position X value
     * @return int
     */
    public int getX() {
        return this.x;
    }

    /**
     * Return position Y value
     * @return int
     */
    public int getY() {
        return this.y;
    }

    /**
     * Print some information about Point 2D
     */
    public void printPoint() {
        System.out.print("Point 2D\n");
        System.out.print("X = " + this.x + "\n");
        System.out.print("Y = " + this.y + "\n");
    }
}
